### DISCUSSION TITLE
You've successfully authenticated, but GitHub does not provide shell access.

### DISCUSSION THREAD

**Discussion by lrpgworkspace** [community/community#33982] [Other Feature Feedback, Questions, & Ideas]
Hi community,

I'm having a problem that Github Shell decided today it wouldn't let me authenticate using my SSH key anymore. 
When using a Windows 10 terminal, and running "ssh -T git@github.com", I'm receiving the following output:
Hi <githhub account name>! You've successfully authenticated, but GitHub does not provide shell access.

When checking my remote with "git remote -v", I'm seeing that I'm using a SSH Url, instead of https:
`$ git remote -v
origin  git@github.com:<githhub account name>/****.git (fetch)
origin  git@github.com:<githhub account name>/****.git (push)
`

When going on the Github repository website, under Settings > SSH and GPG keys, I see my authentication key, so that looks good.
My password hasn't changed, otherwise I wouldn't have been able to use the command "ssh -T git@github.com" in the first place.

The ssh key is also present on disk, under C:\Users\username\.ssh:
id_rsa.pub
id_rsa

I've been trying to google this issue but have not been able to find a solution. Some users have suggested to rebuild the path to the remote, by using the following command (this has not worked for me):
git remote set-url origin git@github.com:<githhub account name>/****.git

Can somebody please assist me with this problem? 

Thanks,
Manuel

**Comment by airtower-luna** [comment]
The message you see when using SSH directly means you correctly added the key to your account. I've seen some cases (both at work and here in the community) of Git and SSH on Windows having different ideas about where the user's home directory is, and thus where to look for keys. :sweat_smile:

I can't say for sure if that's the case for you, but I can tell you how to check: Run Git with [`GIT_SSH_COMMAND`](https://git-scm.com/docs/git#Documentation/git.txt-codeGITSSHCOMMANDcode) set to `ssh -v`, to get debug output from SSH. That'll tell you (among other things) which keys are tried.

**Comment by lrpgworkspace** [comment]
Hi, thanks for your support!

I added GIT_SSH_COMMAND:  ssh -v as a environment variable and ran a "GIT fetch" with it.
I attached the log as a ".txt" file to this post.

I was prompted for my SSH password, the command seemed to have succeeded.
I did find a way that will remember my SSH password, but it works just for the current shell. If I then close and re-open Git Shell, I'm yet again prompted for my SSH password. That defeats the whole purpose of SSH imo.
The 2 commands I have to enter in order to prevent the constant password prompts within the current session are:
eval `ssh-agent -s`	
ssh-add	

Thanks for you help, I would love to get this fixed
[git fetch.txt](https://github.com/community/community/files/9636977/git.fetch.txt)

**Comment by airtower-luna** [comment]
> I added GIT_SSH_COMMAND: ssh -v as a environment variable and ran a "GIT fetch" with it.
> I attached the log as a ".txt" file to this post.

Yes, that looks like the fetch works. That mean means there's some difference to how Git calls SSH by default, but I can't say exactly how. What kind of shell do you use for Git operations (e.g. Git Bash, or Powershell)? It might make a difference.

> The 2 commands I have to enter in order to prevent the constant password prompts within the current session are:
> eval ssh-agent -s
> ssh-add

Using SSH agent is the right approach here. I'm not sure what's the best way to set it up on Windows, maybe a Windows user can chip in here?

As far as SSH is concerned, you can set the [`AddKeysToAgent` option](https://manpages.debian.org/bullseye/openssh-client/ssh_config.5.en.html#AddKeysToAgent) to automatically add the key to the agent the first time it's loaded.

**Reply by lrpgworkspace** [reply]
Hi guys,

I finally was able to fix this, the following thread helped me out:
https://gist.github.com/bsara/5c4d90db3016814a3d2fe38d314f9c23

In short, if you run into this problem, you'll have to add the following 3 files to your user folder:
.ssh/config
.bash_profile
.bashrc

The link above describes the process in detail, hope it helps!

**Reply by Jugarcia01** [reply]
Thank you @lrpgworkspace for the useful info!

**Reply by md1116** [reply]
Solved my issue. Thanks

**Reply by Kouachi-c** [reply]
This info just help me to solve my issue of 3 hours. Thanks !

**Comment by iskode** [comment]
Hello everyone, I am on Mac but none of the posted solutions work for me.
I've created new ssh key but nothing. The strange is that it was working last week.
Since yesterday all of the sudden this error popped up and after wasting +3 hours of reading
what I can get on internet, still the samne problem.
This is a nightmare all collaborative work are blocked.

**Reply by ArmCM** [reply]
me too :(, same in macos

**Comment by 0riv3r** [comment]
In your ~/.ssh/config file you should have:

Host github.com
(add indent spacing) Hostname github.com
(add indent spacing) IdentityFile ~/.ssh/(your github_private_key_name)

**Reply by Gadrawingz** [reply]
Host *
  AddKeysToAgent yes
  UseKeychain yes
  IdentityFile ~/.ssh/id_ed.......

It works fine

**Comment by iskode** [comment]
My content is (as I have 2 github accounts)

```
Host github.com
  AddKeysToAgent yes
  IdentityFile ~/.ssh/iprivate_key_1

Host github.com
  AddKeysToAgent yes
  IdentityFile ~/.ssh/private_key_2

Host github.com
  AddKeysToAgent yes
  UseKeychain yes
```

So i don't have this  `Hostname github.com` ?

**Reply by airtower-luna** [reply]
That's not going to work, as [described in the `ssh_config` documentation](https://manpages.debian.org/bookworm/openssh-client/ssh_config.5.en.html):

> Since the first obtained value for each parameter is used, more host-specific declarations should be given near the beginning of the file, and general defaults at the end.

When connecting to `github.com`, only the `AddKeysToAgent` and `IdentityFile` values from the first section will be used. The trick would be to use *different* `Host` value(s). That's also where `Hostname` comes in, it lets you use a different hostname for the actual connection.

For example, you could have a config like this:

```
Host github.com
  AddKeysToAgent yes
  IdentityFile ~/.ssh/iprivate_key_1

Host github2
  Hostname github.com
  AddKeysToAgent yes
  IdentityFile ~/.ssh/private_key_2
```

This way connecting to `github.com` will use the first key, and if you want to use the second key, you need to connect to `git@github2` (instead of the usual `github.com`, this means you'll have to change the SSH URL for the repository).

**Reply by iskode** [reply]
Thank you so much.

I have seen this [post](https://gist.github.com/oanhnn/80a89405ab9023894df7) which solves my multi user/key problem.
Hope it helps.

**Comment by shahabijaz1997** [comment]
<img width="790" alt="cap prod error" src="https://github.com/community/community/assets/145776313/5d5dbfbd-2dd9-45a9-87ab-ee8ceaaadbef">

I'm facing this error where it says repository not found but it did authenticate it when I ran **ssh -T git@github.com**. Can anyone please provide a possible solution and why it's happening.

**Comment by parsley72** [comment]
It looks like this is expected:
https://docs.github.com/en/authentication/connecting-to-github-with-ssh/testing-your-ssh-connection

**Reply by kathyrollo** [reply]
This is probably the right answer.

**Comment by mohammedbabiker** [comment]
its properly you use different branch that main one, Keep in mind that you can **pull** from GitHub but you can't **push** so you need to set remote url for the branch you want to working on with ``` git remote set-url origin git@github.com:<username>/<repo>.git``` and it should work just fine.

**Reply by surajsharma** [reply]
this worked! but do i have to do this for each repo?

**Reply by Hryhorii83** [reply]
Ok?

**Comment by JulesC836** [comment]
I found an easy solution here: https://stackoverflow.com/a/26955444
worked for me

**Comment by MarsYoung** [comment]
In my case, it happens when I enabled the deploy key in Orgnization settings, then the personal key not work on the orgnization repos.
The fix is disable the deploy key.
